### ISSUE TITLE
Orchestrator:prepare not finding all artifacts, despite them being available and published

### ISSUE THREAD

**Issue by dschibster** [flxbl-io/sfp#15]
**Describe the bug**
While trying to build a dev pool, prepare fetches a list of all artifacts to be deployed from our npm registry. All other CICD operations, e.g. validate and build, are able to retrieve the artifact without a problem. However, the prepare step does not find two of our artifacts, and it is always the same two. I have tried fixing a version in my release definition, to no avail. I have also tried using a release config file, but that also did not help out here.

**To Reproduce**
For me, it's as simple as trying to run `sfp orchestrator:prepare -f config/sfdx-pool/dev/dev-pool-def-1.json -v devhub --npmrcpath $(npmrc.secureFilePath) --loglevel trace` in our pipeline.

What I get is the following log

```
Fetching src-env-specific-pre using npm pack @customerproject/src-env-specific-pre_sfpowerscripts_artifact@5.0.0-82450
Fetching utility using npm pack @customerproject/utility_sfpowerscripts_artifact@0.1.0-170
Fetching salescloud using npm pack @customerproject/salescloud_sfpowerscripts_artifact@0.1.0-192
Artifact  for salescloud missing in NPM Registry provided, This might result in deployment failures
Fetching salescloudb2b using npm pack @customerproject/salescloudb2b_sfpowerscripts_artifact@0.1.0-189
Fetching servicecloud using npm pack @customerproject/servicecloud_sfpowerscripts_artifact@0.1.0-189
Fetching marketing-consent using npm pack @customerproject/marketing-consent_sfpowerscripts_artifact@0.1.0-186
Fetching experiencecloud using npm pack @customerproject/experiencecloud_sfpowerscripts_artifact@0.1.0-174
Fetching analytics using npm pack @customerproject/analytics_sfpowerscripts_artifact@0.1.0-23
Fetching src-vlocity-management using npm pack @customerproject/src-vlocity-management_sfpowerscripts_artifact@9.0.0-82659
Fetching vlocitycustomer using npm pack @customerproject/vlocitycustomer_sfpowerscripts_artifact@4.0.0-82659
Artifact  for vlocitycustomer missing in NPM Registry provided, This might result in deployment failures
Fetching customerui using npm pack @customerproject/customerui_sfpowerscripts_artifact@11.0.0-82659
Fetching src-access-management using npm pack @customerproject/src-access-management_sfpowerscripts_artifact@8.0.0-82659
```

It's always the same two artifacts that cannot be found, although I have checked multiple times that yes, they do in fact exist, and out build pipeline also successfully deployed both of them to our UAT instance just half an hour ago. The problem is exclusive to `prepare` as far as I can tell.

As the other artifacts can be found, this also seems to be not an issue with npm authentication. I would expect nothing to be found in that way.

**Expected behavior**
All artifacts should be found and installed.

**Platform Details (please complete the following information):**
This is using the latest release of the sfpowerscripts repository container. (25.5.3-7028186930)

**Additional context**
I get that this is probably more of an npm issue than an sfp one. If I can give any more information or get pointers on what to check additionally, I would greatly appreciate it.

**Comment by azlam-abdulsalam** [comment]
I think i have updated the error handling to provide more details when this happens as part of another issue as part of the next release. I will double check it 

Meanwhile can you try npm pack command directly?

Also what's rhe repository provider you are targeting? Could it be an issue with scope?

**Comment by dschibster** [comment]
We are using Artifacts inside Azure Devops to save our artifacts. Regarding using npm pack directly I will have to check how to directly authenticate, as this part of the pipeline was configured by another colleague. Some more outputs in the logs would be a great start!

**Comment by Rocko1204** [comment]
Hi @dschibster Can you please check one thing for me. Is the version from the screenshot really the latest version/tag for the package? You should know that we use a different background process for the prepare job to identify the package version. We use git to search for the last tag for the package. So is salescloud 0.1.0-192 really the latest version and have you really exactly this artifacts in the registry? It can happen with wrong job settings (wrong cloning strategy or git fetch tags settings) that not all tags are found and thus a wrong reference to the artifacts is created.

**Comment by dschibster** [comment]
I have one update!
It seems like the console output is wrong, or that prepare has a fallback, because following the dev pool creation, although the artifacts were not found they do in fact get installed and working. The installation errors I was facing came from malconfigured packages.

**Comment by azlam-abdulsalam** [comment]
@dschibster New clear log messages are added and fixed in Feb2024 versions. Please do let me know if it is helping you.
